reactome_gnn/network.py

```diff
@@ -96,16 +112,45 @@ def get_name_to_id(self):
             self.name_to_id[info.name] = id
         return self.name_to_id
 
-    def to_networkx(self, type='txt'):
+    def to_networkx(self, type='json'):
         self.graph_nx = nx.DiGraph()
+        stids, names, parents, weights = {}, {}, {}, {}
         graph = self.graph_txt if type == 'txt' else self.graph_dict
         for key, values in graph.items():
             for value in values:
                 self.graph_nx.add_edge(key, value)
+
+        # TODO: Optimize this, too many of those values are calculated multiple times
+        for key, value in self.node_attributes.items():
+            stids[key] = value['stid']
+            names[key] = value['name']
+            parents[key] = value['parent']
+            weights[key] = value['weight']
+
+        nx.set_node_attributes(self.graph_nx, stids, 'stid')
+        nx.set_node_attributes(self.graph_nx, names, 'name')
+        nx.set_node_attributes(self.graph_nx, parents, 'parent')
+        nx.set_node_attributes(self.graph_nx, weights, 'weight')
         return self.graph_nx
 
     def visualize(self, fig_path='data/graph.png'):
         if self.graph_nx is None:
             self.to_networkx()
         nx.draw(self.graph_nx, node_size=2, width=0.2, arrowsize=2)
-        plt.savefig(fig_path)
\ No newline at end of file
+        plt.savefig(fig_path)
+
+    def enrichment_analysis(self):        
+        result = analysis.identifiers(ids=self.markers, interactors=False, page_size='1', page='1', species='Homo Sapiens', sort_by='ENTITIES_FDR', order='ASC', resource='TOTAL', p_value='1', include_disease=False, min_entities=None, max_entities=None, projection=True)
+        token = result['summary']['token']
+        token_result = analysis.token(token, species='Homo sapiens', page_size='-1', page='-1', sort_by='ENTITIES_FDR', order='ASC', resource='TOTAL', p_value='1', include_disease=False, min_entities=None, max_entities=None)
+        return [p['stId'] for p in token_result['pathways']]
+
+    def set_weights(self, mode):
+        assert mode in (1, 2, 3), "Mode has to be 1, 2, or 3."
+        if mode == 1:
+            stids = self.enrichment_analysis()
+        if mode == 2:
+            stids = fiviz.ehld_stids()
```

Comment:
The modes need to evaluate the enrichment analysis using the total pathways in (EHLD or SBGN each separately) and the users markers.

Reply:
So in mode 1 enrichment analysis is done on all the pathways (nodes in the graph), whereas in modes 2 and 3 it is done only on those that are in EHLD/SBGN, respectively?

Reply:
If that is correct, then should I always first perform EA on all the nodes and then, e.g., in case EHLD, make an intersection between all the obtained pathways (as in mode=1) and the ones in the EHLD?